### ISSUE TITLE
About project LICENSE

### ISSUE THREAD

**Issue by matshou** [AlbertTheTerrible/Mushroom-Dream#14]
The project license imposes the following restrictions to users:

> Any clone, variation or other distribution of the game has NO permission to use any of the work in any way. You do not have any permission to distribute, rehost, relicense, sell for any currency of any kind (directly or indirectly), re-name, use textures/images from, edit or modify the above mentioned works in the tileset or any of it's variations/modifications (no matter how much it changed) for any reason unless you get written and dated permission from me.

These restrictions are in violation of the implicit license granted to users by [GitHub Terms of Service](https://docs.github.com/en/github/site-policy/github-terms-of-service#5-license-grant-to-other-users):

> If you set your pages and repositories to be viewed publicly, you grant each User of GitHub a nonexclusive, worldwide license to use, display, and perform Your Content through the GitHub Service and to reproduce Your Content solely on GitHub as permitted through GitHub's functionality (for example, through forking).

By hosting your project on GitHub you are by default giving others the following permissions:

- Allowed to *use* and *reproduce* by cloning through GitHub Service.
- Allowed to *re-host* by creating a project fork on GitHub.
- Allowed to *distribute* on their own project fork on GitHub.

As far as I understand the only valid license restriction of your license is the prohibition of derivative work.

User contributions are also worth consideration. I don't know if you accept art contributions to the project but as it currently stands users are not encouraged to contribute to the project as the legality of who owns what gets a bit murky when dealing with copyright licenses. You should perhaps explore other [license options](https://choosealicense.com/) that are more compatible with open source development.

In conclusion I would recommend amending the license to be more permissive and thus compatible with the implicit user license granted by GitHub and also explore the possibility of changing the license to a more permissive model as a whole.

**Comment by AlbertTheTerrible** [comment]
Thank you for bringing that up, i honestly had no idea of those terms on github. My terms are final, as i really have no interest in my work or name being used for anything else other than the CDDA project, so hosting the project here might be incompatible.

Also i am not really looking for any contributions outside bugfixes. Outside the mashup, all art contributions since the inception of the project are made by me and i'd like them to remain as such. 

If possible i was thinking of keeping the page, but hosting the project itself elsewhere.

**Comment by matshou** [comment]
> If possible i was thinking of keeping the page, but hosting the project itself elsewhere.

I would recommend still keeping the project on GitHub but moving the actual assets to a private repository so you still benefit from Git workflow while keeping the assets under full copyright. The current repository could be used to publish releases although I don't know how the GitHub user license applies to them.

**Comment by AlbertTheTerrible** [comment]
Thanks! That's a great suggestion. I'm gonna have to take a look at how i could work with a private version of this

**Comment by AlbertTheTerrible** [comment]
I've updated the page to hopefully fix the collision with the licensing. I'm still gonna see what i'm gonna do with the releases. Thank you so much for pointing this out 👍

**Comment by AlbertTheTerrible** [comment]
I suck at these things, i should start reading the terms and conditions 😁

**Comment by matshou** [comment]
No problem, don't hesitate to ask for any help you might need with these matters in the future.

**Comment by AlbertTheTerrible** [comment]
Just to make my stance clear on this and for future reference too. I am making this tileset for the CDDA game distributed under CleverRaven and it's community, and the license I wrote reflects this. I have no interest of having any of it's parts being used anywhere else that is unrelated to it. 
If someone is taking CDDA in a different direction, or managing a different distribution (for example, a distribution for mobile devices) and is interested in using the works here to fill the project, you can always leave me a message and I will look into it. I can easily write a custom license to those projects with no issue provided the following happens:

- Those projects respect the licensing here
- They do not direct hate to any individual or group of people
- The project adds value to the community
- They do not charge money or any currency (donations are fine)
- No involvement with any sort of scams, malware or cryptocurrencies/nfts

**Comment by waveyl** [comment]
Can I (or someone else) upload this tilesets to QQ group (a chat tool similar to Telegram) without any modifications?

**Comment by waveyl** [comment]
The reason is it's difficult for some of us to download from github.

**Comment by matshou** [comment]
Can you explain why does downloading from Github provide difficulties for you?

**Comment by waveyl** [comment]
Sometimes we can't access Github properly because of government policy

**Comment by ZhilkinSerg** [comment]
Tileset downloads are hosted on Patreon, not on Github.

**Comment by AlbertTheTerrible** [comment]
Hey there, sorry for the late reply. I didn't get a notification to the replies on this post. The current license does not allow rehosting anywhere else w/o permission and like ZhilkinSerg mentioned, the tileset is currently hosted on Patreon. Is patreon also an issue in terms of accessibility?

**Comment by waveyl** [comment]
No, Patreon is OK. Thanks all of you for clearing up my confusion.
